core/lib/utils/log_tracking.dart

```diff
@@ -0,0 +1,192 @@
+import 'dart:async';
+import 'dart:collection';
+import 'dart:io';
+
+import 'package:core/domain/exceptions/download_file_exception.dart';
+import 'package:core/utils/app_logger.dart';
+import 'package:core/utils/file_utils.dart';
+import 'package:core/utils/platform_info.dart';
+import 'package:dartz/dartz.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter/foundation.dart';
+import 'package:intl/intl.dart';
+import 'package:path_provider/path_provider.dart';
+
+class LogTracking {
+  static const String logFolder = 'TraceLogs';
+  static const String logFileNameDatePattern = 'yyyy-MM-dd';
+  static const String logMessageDatePattern = 'yyyy-MM-dd, HH:mm:ss';
+
+  LogTracking._();
+
+  factory LogTracking() => _instance ??= LogTracking._();
+
+  static LogTracking? _instance;
+
+  bool enableTraceLog = PlatformInfo.isMobile;
+
+  final Queue<String> _messagesQueue = Queue();
+  bool _isScheduled = false;
+
+  Future<void> addLog({required String message}) async {
+    _messagesQueue.add(message);
+
+    if (!_isScheduled) {
+      _isScheduled = true;
+      await _executeTraceLog();
+    }
+  }
+
+  Future _executeTraceLog() async {
+    while (true) {
+      if (_messagesQueue.isEmpty) {
+        _isScheduled = false;
+        return;
+      }
+
+      final message = _messagesQueue.removeFirst();
+      await saveLog(message: message);
```

Comment:
can we need try-catch here?

Reply:
IMO, We don't need to, because the possibility of errors here is almost non-existent. If there is only in the `saveLog` function, we have `try/catch` it.

Reply:
try/catch added